exams/api.py

```diff
@@ -130,28 +129,30 @@ def authorize_for_exam_run(mmtrack, course_run, exam_run):
 
 def authorize_for_latest_passed_course(mmtrack, exam_run):
     """
-    This walks the CachedEnrollments backwards chronologically and authorizes the first eligible one.
+    This walks the FinalGrade backwards chronologically and authorizes the first eligible one.
 
     Args:
         mmtrack (dashboard.utils.MMTrack): An instance of all user information about a program
         exam_run (exams.models.ExamRun): the ExamRun to authorize the learner for
     """
-    enrollments_qset = CachedEnrollment.objects.filter(
+    final_grades = FinalGrade.objects.filter(
         user=mmtrack.user,
+        passed=True,
```

Comment:
Why did you add `passed=True` to the filter?

Reply:
Dont we need only authorize for passed run? https://github.com/mitodl/micromasters/blob/master/exams/api.py#L104

I am filtering passed user above.